pkg/crc/api/handlers.go

```diff
@@ -26,6 +31,7 @@ func NewHandler(config crcConfig.Storage, machine machine.Client) *Handler {
 			Underlying: machine,
 		},
 		Config: config,
+		lock:   semaphore.NewWeighted(int64(1)),
```

Comment:
No, functionally a mutex would be enough, but i wanted the `TryAcquire() bool` method to check if the lock can be attained, which is not there in `Mutex`

Reply:
Ah right, no `TryLock()`, apparently this is on purpose. Seems like the go way of handling this would be to use `select`+go channel+ another go routine